tensorboard/plugins/hparams/summary.py

```diff
@@ -0,0 +1,144 @@
+   summary into the same session run <session_name>.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import datetime
+
+import six
+import tensorflow as tf
+from google.protobuf import timestamp_pb2
+
+from tensorboard.plugins.hparams import api_pb2
+from tensorboard.plugins.hparams import plugin_data_pb2
+from tensorboard.plugins.hparams import metadata
+
+
+def experiment_pb(
+    hparam_infos,
+    metric_infos,
+    user="",
+    description="",
+    time_created=datetime.datetime.now()):
+  """Creates a summary that defines a hyperparameter-tuning experiment.
+  Arguments:
+    hparam_infos: Array of api_pb2.HParamInfo messages. Describes the
+        hyperparameters used in the experiment.
+    metric_infos: Array of api_pb2.MetricInfo messages. Describes the metrics
+        used in the experiment. See the documentation at the top of this file
+        for how to populate this.
+    user: String. An id for the user running the experiment
+    description: String. A description for the experiment. May contain markdown.
+    time_created: datetime. The time the experiment is created.
+
+  Returns:
+    A summary protobuffer containing the experiment definition.
+  """
+  experiment = api_pb2.Experiment(
+      description=description,
+      user=user,
+      time_created=timestamp_pb2.Timestamp().FromDatetime(time_created),
+      hparam_infos=hparam_infos,
+      metric_infos=metric_infos)
+  return _summary(metadata.EXPERIMENT_TAG,
+                  plugin_data_pb2.HParamsPluginData(experiment=experiment))
+
+
+def session_start_pb(hparams,
+                     model_uri="",
+                     monitor_url="",
+                     group_name=""):
+  """Creates a summary that contains a training session metadata information.
+  One such summary per training session should be created. Each should have
+  a different run.
+
+  Arguments:
+    hparams: A dictionary with string keys. Describes the hyperparameter values
+             used in the session mappng each hyperparameter name to its value.
+             Supported value types are  bool, int, float, or str.
+    model_uri: See the comment for the field with the same name of
+               plugin_data_pb2.SessionStartInfo.
+    monitor_url: See the comment for the field with the same name of
+                 plugin_data_pb2.SessionStartInfo.
+    group_name:  See the comment for the field with the same name of
+                 plugin_data_pb2.SessionStartInfo.
+  Returns:
+    Returns the summary protobuffer mentioned above.
+  """
+  session_start_info = plugin_data_pb2.SessionStartInfo(
+      model_uri=model_uri,
+      monitor_url=monitor_url,
+      group_name=group_name)
+  for (hp_name, hp_val) in six.iteritems(hparams):
+    if isinstance(hp_val, (float, int)):
+      session_start_info.hparams[hp_name].number_value = hp_val
+    elif isinstance(hp_val, str, unicode):
```

Comment:
Pylint is complaining that `unicode` is not defined in python 3.

I'm not quite sure whether the python protobuf API under python 3 will accept str or unicode or both for string_value, it might be worth checking to see how it behaves.  Depending on the results you probably want the second argument to is_instance to be either `six.binary_type` (str on py2, bytes on py3), `six.string_types` (str or unicode on py2, str on py3), or `(six.binary_type, six.text_type)` (str or unicode on py2, bytes or str on py3).

I won't insist on this, but given that this logic isn't entirely straightforward, it might be worth having a summary_test.py with a few tests to make sure the expected Summary objects are being produced for the given function calls here.  Could be a follow-up PR.

Reply:
Done. The protobuf docs suggest that a proto string is a unicode UTF-8 encoded string and it accepts both unicode and str objects.
I'll add a test in a follow-up PR.